src/Microsoft.VisualStudio.ProjectSystem.Managed/GlobalSuppressions.cs

```diff
@@ -1,3 +1,5 @@
 ﻿[assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("Usage", "VSTHRD110:Observe result of async calls", Justification = "https://github.com/dotnet/project-system/issues/3921", Scope = "member", Target = "~M:Microsoft.VisualStudio.ProjectSystem.JoinableTaskQueue.Register(Microsoft.VisualStudio.Threading.JoinableTask)")]
 [assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("Usage", "VSTHRD004:Await SwitchToMainThreadAsync", Justification = "https://github.com/Microsoft/vs-threading/issues/377", Scope = "member", Target = "~M:Microsoft.VisualStudio.ProjectSystem.ThreadingExtensions.SwitchToUIThread(Microsoft.VisualStudio.ProjectSystem.IProjectThreadingService,System.Threading.CancellationToken)~Microsoft.VisualStudio.Threading.JoinableTaskFactory.MainThreadAwaitable")]
 [assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("Reliability", "CA2000:Dispose objects before losing scope", Scope = "member", Target = "~M:Microsoft.VisualStudio.Threading.Tasks.CancellationSeries.CreateNext(System.Threading.CancellationToken)~System.Threading.CancellationToken")]
+[assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("ApiDesign", "RS0030:Do not used banned APIs", Justification = "<Pending>", Scope = "type", Target = "~T:Microsoft.VisualStudio.ProjectSystem.ISafeProjectGuidService")]
```

Comment:
Do we have guidance on when we use a suppression attribute vs. when we use `#pragma warning'? We have a mix of both through the code.

Reply:
I don't get why this was added, why is the rule firing?

Reply:
Both of these suppressions are replacements for `#pragma warning` items that've been removed from other files.

In both cases it's due to legitimate usages of the banned symbol. The 'safe' implementation of this interface needs to use the actual interface, so gets a pass.

I'm undecided on whether it's clearer to have the suppression here or alongside the code in question.

Reply:
> why is the rule firing

The rule's firing in `ISafeProjectGuidService.cs` due to the reference of `IProjectGuidService` in the XML docs.

Reply:
I use suppression files when; 1 it's a bug in the rule, 2 I'm baselining. I'll usually inline the rest.

Reply:
I think that's a sensible approach. Under those guidelines these'd be inline.

Reply:
My thought was basically:

- If the rule is likely to be re-enabled at some point on that code then use `#pragma` suppression so it's easy to search for these

- If a rule is being suppressed because we don't want to change / will never change the code and it's not a bug then I think moving it to a suppression file and not having `#pragma`'s clutter things up is nice

Don't have any strong feelings about this though.  If someone has a preference here I will revert it.